### PR TITLE
Modify test-app to check typescript build too

### PR DESCRIPTION
**Pull request by jvanbruegge**
ONLY MERGE AFTER #86 HAS BEEN PUBLISHED ON NPM

Adds a TsApp React component written in TSX, adds the typescript block to the webpack config.
Tests will fail initially, should pass once the typescript block is on npm.

### CODE REVIEW

File: test-app/package.json [hunk 1]
```diff
@@ -7,8 +7,11 @@
     "node": ">= 5.0"
   },
   "devDependencies": {
+    "@types/node": "^7.0.0",
+    "@types/react": "^15.0.0",
     "@webpack-blocks/babel6": "^0.x",
     "@webpack-blocks/css-modules": "^0.x",
+    "@webpack-blocks/typescript": "^0.x",
```
Comment (andywer) [inline test-app/package.json]: Please change to a proper version number. I already changed it for `babel6` and `css-modules` in the release branch :)

File: test-app/src/TsApp.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import * as React from 'react'
+const styles = require('./App.css')
+
+const TsApp : (() => JSX.Element) = () =>
```
Comment (andywer) [inline test-app/src/TsApp.tsx]: I suggest we go all the way here. If we have a TypeScript-based example then let's drop the JS one.
And we can rename `TsApp` => `App` then 😉

**Comment by andywer** [comment]
A `git fetch && git rebase origin/typescript` might be a good idea here. It's based on the other PR, but not yet up-to-date in respect of its recent changes 😉

**Comment by SuperManitu** [comment]
rebased on the current master

**Comment by andywer** [comment]
One question: I tried it locally and it works nicely. But it creates a `App.js` and `App.js.map` file in `test-app/src/`. Can we make it only write those to the output bundle instead of the `src/` directory? Or at least in a `lib/` directory?

**Comment by jvanbruegge** [comment]
This should only happen if you run tsc over the directory. For me running `npm run build` just updates the bundle.js (and the css/html)

**Comment by andywer** [comment]
Ahh, sorry, nevermind! Just installed `atom-typescript` and it has this "compileOnSave" feature enabled by default 🙈😉

**Comment by jvanbruegge** [comment]
you can add `"compileOnSave": false` to the `tsconfig.json`

**Comment by andywer** [comment]
Yeah, that looks like a good idea! 👍

**Comment by andywer** [comment]
Did you write the `tsconfig.json` by hand or is it generated by some other tool? (Since it's the only file with 4 spaces indent)

**Comment by jvanbruegge** [comment]
Basicly copy & paste it over and over, my default indent is 4 spaces. It's no problem to change that

**Comment by andywer** [comment]
Wow, that was fast 😉

**Comment by jvanbruegge** [comment]
Just edited in Github

**Comment by andywer** [comment]
I thought so, but still... Will merge it after Travis is done running the tests :)

**Comment by jvanbruegge** [comment]
Cool 😄

**Pull request merged**